server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -172,10 +172,14 @@ public Invocations getInvocations() {
          responseBuilder.status(HttpResponseStatus.NOT_FOUND);
          return CompletableFuture.completedFuture(responseBuilder.build());
       }
-      responseBuilder.entity(new CacheInputStream(cache.keySet().stream(), batch));
+      // Streaming over the cache is blocking
```

Comment:
Could we use `cache.keySet().localPublisher(null)` to make it non-blocking?

Reply:
The problem is that method doesn't yet support encoding. I would love to add that at some point, I was originally waiting for the new API to do that. But I think I will have to add that for 11.1 as it is becoming more and more needed.

Reply:
The method is really designed for local invocation only currently and isn't used in the remote part and is actually blocking currently :(